src/app/components/navigation/navigation.component.js

```diff
@@ -1,13 +1,31 @@
 const template = require('./navigation.html');
 
 class NavigationController {
-  constructor($state) {
+  constructor($state, ExpeditionService) {
     'ngInject';
 
     this.$state = $state;
+    this.ExpeditionService = ExpeditionService;
   }
 
-  $onChanges() {
+  $onChanges(changesObj) {
+    if (
+      this.currentUser &&
+      this.currentProject &&
+      'currentProject' in changesObj
+    ) {
+      this.ExpeditionService.getExpeditionsForUser(
+        this.currentProject.projectId,
+        true,
+      ).then(({ data }) => {
+        this.data = data;
```

Comment:
I don't think you need this.data, you can simplify this whole thing to  `this.showMyExpeditions = data.length > 0`